Add static-check analysis to the test-script,.

Some code-purity issues flagged:

```
cmd_metrics.go:64:22: should omit comparison to bool constant, can be simplified to !nop (S1002)
cmd_metrics_test.go:71:12: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (SA1012)
cmd_serve.go:76:19: error strings should not be capitalized (ST1005)
cmd_serve.go:89:19: error strings should not be capitalized (ST1005)
cmd_serve.go:256:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:275:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:316:19: error strings should not be capitalized (ST1005)
cmd_serve.go:358:29: the argument is already a string, there's no need to use fmt.Sprintf (S1025)
cmd_serve.go:377:45: the argument is already a string, there's no need to use fmt.Sprintf (S1025)
cmd_serve.go:385:19: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:417:19: error strings should not be capitalized (ST1005)
cmd_serve.go:432:19: error strings should not be capitalized (ST1005)
cmd_serve.go:477:20: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:496:20: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:541:19: error strings should not be capitalized (ST1005)
cmd_serve.go:551:19: error strings should not be capitalized (ST1005)
cmd_serve.go:619:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:659:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:705:19: error strings should not be capitalized (ST1005)
cmd_serve.go:784:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:815:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:854:20: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:890:20: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:994:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_serve.go:1013:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
cmd_version_test.go:20:12: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (SA1012)
cmd_version_test.go:37:12: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (SA1012)
db.go:247:24: error strings should not be capitalized (ST1005)
db.go:462:4: should merge variable declaration with assignment on next line (S1021)
db.go:588:26: error strings should not be capitalized (ST1005)
db.go:633:27: error strings should not be capitalized (ST1005)
static.go:124:24: error strings should not be capitalized (ST1005)
yaml_parser.go:118:20: error strings should not be capitalized (ST1005)
yaml_parser.go:126:20: error strings should not be capitalized (ST1005)
yaml_parser.go:144:20: error strings should not be capitalized (ST1005)
yaml_parser.go:173:20: error strings should not be capitalized (ST1005)
yaml_parser.go:280:20: error strings should not be capitalized (ST1005)
yaml_parser.go:317:20: error strings should not be capitalized (ST1005)
yaml_parser.go:408:23: error strings should not be capitalized (ST1005)
yaml_parser_test.go:98:3: should merge variable declaration with assignment on next line (S1021)
yaml_parser_test.go:165:3: should merge variable declaration with assignment on next line (S1021)
```

--- a/.github/run-tests.sh
+++ b/.github/run-tests.sh
@@ -1,8 +1,21 @@
 #!/bin/sh
 
-# Install the lint-tool, and the shadow-tool
+# Install tools to test our code-quality.
 go get -u golang.org/x/lint/golint
 go get -u golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
+go get -u honnef.co/go/tools/cmd/staticcheck
+
+
+# Run the static-check tool.
+t=$(mktemp)
+staticcheck -checks all ./... | grep -v context.TODO > $t
+if [ -s $t ]; then
+    echo "Found errors via 'staticcheck'"
+    cat $t
+    rm $t
+    exit 1
+fi
+rm $t
 
 # At this point failures cause aborts
 set -e

--- a/cmd_metrics.go
+++ b/cmd_metrics.go
@@ -11,7 +11,7 @@ import (
 	"os"
 
 	"github.com/google/subcommands"
-	"github.com/marpaia/graphite-golang"
+	graphite "github.com/marpaia/graphite-golang"
 )
 
 //
@@ -61,7 +61,7 @@ func SendMetrics(host string, port int, prefix string, nop bool) {
 	// If there was an error in the helper we're OK,
 	// providing we are running in `-nop`-mode.
 	//
-	if (err != nil) && (nop == false) {
+	if (err != nil) && (!nop) {
 		fmt.Printf("Error creating metrics-helper: %s\n", err.Error())
 		return
 	}

--- a/cmd_serve.go
+++ b/cmd_serve.go
@@ -73,7 +73,7 @@ func APIState(res http.ResponseWriter, req *http.Request) {
 	//
 	if len(state) < 1 {
 		status = http.StatusNotFound
-		err = errors.New("Missing 'state' parameter")
+		err = errors.New("missing 'state' parameter")
 		return
 	}
 
@@ -86,7 +86,7 @@ func APIState(res http.ResponseWriter, req *http.Request) {
 	case "failed":
 	case "orphaned":
 	default:
-		err = errors.New("Invalid state")
+		err = errors.New("invalid state supplied")
 		status = http.StatusInternalServerError
 		return
 	}
@@ -253,7 +253,7 @@ func RadiatorView(res http.ResponseWriter, req *http.Request) {
 		//
 		tmpl, err := getResource("data/radiator.template")
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -272,7 +272,7 @@ func RadiatorView(res http.ResponseWriter, req *http.Request) {
 		//
 		// If there were errors, then show them.
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -313,7 +313,7 @@ func ReportSubmissionHandler(res http.ResponseWriter, req *http.Request) {
 	// Ensure this was a POST-request
 	//
 	if req.Method != "POST" {
-		err = errors.New("Must be called via HTTP-POST")
+		err = errors.New("must be called via HTTP-POST")
 		status = http.StatusInternalServerError
 		return
 	}
@@ -355,7 +355,7 @@ func ReportSubmissionHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	// (Which is something you might do when testing the dashboard.)
 	//
-	path := filepath.Join(dir, fmt.Sprintf("%s", report.Hash))
+	path := filepath.Join(dir, report.Hash)
 
 	if Exists(path) {
 		fmt.Fprintf(res, "Ignoring duplicate submission")
@@ -374,7 +374,7 @@ func ReportSubmissionHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	// Record that report in our SQLite database
 	//
-	relativePath := filepath.Join(report.Fqdn, fmt.Sprintf("%s", report.Hash))
+	relativePath := filepath.Join(report.Fqdn, report.Hash)
 
 	addDB(report, relativePath)
 
@@ -382,7 +382,7 @@ func ReportSubmissionHandler(res http.ResponseWriter, req *http.Request) {
 	// Show something to the caller.
 	//
 	out := fmt.Sprintf("{\"host\":\"%s\"}", report.Fqdn)
-	fmt.Fprintf(res, string(out))
+	fmt.Fprint(res, string(out))
 
 }
 
@@ -414,7 +414,7 @@ func SearchHandler(res http.ResponseWriter, req *http.Request) {
 	// Ensure this was a POST-request
 	//
 	if req.Method != "POST" {
-		err = errors.New("Must be called via HTTP-POST")
+		err = errors.New("must be called via HTTP-POST")
 		status = http.StatusInternalServerError
 		return
 	}
@@ -429,7 +429,7 @@ func SearchHandler(res http.ResponseWriter, req *http.Request) {
 	// Ensure we have a term.
 	//
 	if len(term) < 1 {
-		err = errors.New("Missing search term")
+		err = errors.New("missing search term")
 		status = http.StatusInternalServerError
 		return
 	}
@@ -474,7 +474,7 @@ func SearchHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	tmpl, err := getResource("data/results.template")
 	if err != nil {
-		fmt.Fprintf(res, err.Error())
+		fmt.Fprint(res, err.Error())
 		return
 	}
 
@@ -493,7 +493,7 @@ func SearchHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	// If there were errors, then show them.
 	if err != nil {
-		fmt.Fprintf(res, err.Error())
+		fmt.Fprint(res, err.Error())
 		return
 	}
 
@@ -538,7 +538,7 @@ func ReportHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	if len(id) < 1 {
 		status = http.StatusNotFound
-		err = errors.New("Missing 'id' parameter")
+		err = errors.New("missing 'id' parameter")
 		return
 	}
 
@@ -548,7 +548,7 @@ func ReportHandler(res http.ResponseWriter, req *http.Request) {
 	reg, _ := regexp.Compile("^([0-9]+)$")
 	if !reg.MatchString(id) {
 		status = http.StatusInternalServerError
-		err = errors.New("The report ID must be numeric")
+		err = errors.New("the report ID must be numeric")
 		return
 	}
 
@@ -616,7 +616,7 @@ func ReportHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		tmpl, err := getResource("data/report.template")
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -656,7 +656,7 @@ func ReportHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		// If there were errors, then show them.
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -702,7 +702,7 @@ func NodeHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	if len(fqdn) < 1 {
 		status = http.StatusNotFound
-		err = errors.New("Missing 'fqdn' parameter")
+		err = errors.New("missing 'fqdn' parameter")
 		return
 	}
 
@@ -781,7 +781,7 @@ func NodeHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		tmpl, err := getResource("data/node.template")
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -812,7 +812,7 @@ func NodeHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		// If there were errors, then show them.
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -851,7 +851,7 @@ func IconHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	data, err := getResource("data/favicon.ico")
 	if err != nil {
-		fmt.Fprintf(res, err.Error())
+		fmt.Fprint(res, err.Error())
 		return
 	}
 
@@ -887,7 +887,7 @@ func SorterHandler(res http.ResponseWriter, req *http.Request) {
 	//
 	data, err := getResource("data/jquery.tablesorter.min.js")
 	if err != nil {
-		fmt.Fprintf(res, err.Error())
+		fmt.Fprint(res, err.Error())
 		return
 	}
 
@@ -991,7 +991,7 @@ func IndexHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		tmpl, err := getResource("data/index.template")
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 
@@ -1010,7 +1010,7 @@ func IndexHandler(res http.ResponseWriter, req *http.Request) {
 		//
 		// If there were errors, then show them.
 		if err != nil {
-			fmt.Fprintf(res, err.Error())
+			fmt.Fprint(res, err.Error())
 			return
 		}
 

--- a/cmd_serve_test.go
+++ b/cmd_serve_test.go
@@ -47,7 +47,7 @@ func TestNonNumericReport(t *testing.T) {
 		}
 
 		// Check the response body is what we expect.
-		expected := "The report ID must be numeric\n"
+		expected := "the report ID must be numeric\n"
 		if rr.Body.String() != expected {
 			t.Errorf("handler returned unexpected body: got '%v' want '%v'",
 				rr.Body.String(), expected)
@@ -95,7 +95,7 @@ func TestUknownAPIState(t *testing.T) {
 		if status := resp.StatusCode; status != http.StatusInternalServerError {
 			t.Errorf("Unexpected status-code: %v", status)
 		}
-		if content != "Invalid state\n" {
+		if content != "invalid state supplied\n" {
 			t.Fatalf("Unexpected body: '%s'", body)
 		}
 	}
@@ -371,7 +371,7 @@ func TestSearchMethod(t *testing.T) {
 	}
 
 	// Check the response body is what we expect.
-	expected := "Must be called via HTTP-POST\n"
+	expected := "must be called via HTTP-POST\n"
 	if rr.Body.String() != expected {
 		t.Errorf("handler returned unexpected body: got '%v' want '%v'",
 			rr.Body.String(), expected)
@@ -403,7 +403,7 @@ func TestSearchEmpty(t *testing.T) {
 	}
 
 	// Check the response body is what we expect.
-	expected := "Missing search term\n"
+	expected := "missing search term\n"
 	if rr.Body.String() != expected {
 		t.Errorf("handler returned unexpected body: got '%v' want '%v'",
 			rr.Body.String(), expected)
@@ -485,7 +485,7 @@ func TestUploadReportMethod(t *testing.T) {
 	}
 
 	// Check the response body is what we expect.
-	expected := "Must be called via HTTP-POST\n"
+	expected := "must be called via HTTP-POST\n"
 	if rr.Body.String() != expected {
 		t.Errorf("handler returned unexpected body: got '%v' want '%v'",
 			rr.Body.String(), expected)
@@ -603,7 +603,7 @@ func TestUploadBogusReport(t *testing.T) {
 	}
 
 	// Check the response body is what we expect.
-	expected := "Failed to get 'host' from YAML\n"
+	expected := "failed to get 'host' from YAML\n"
 	if rr.Body.String() != expected {
 		t.Errorf("handler returned unexpected body: got '%v' want '%v'",
 			rr.Body.String(), expected)

--- a/db.go
+++ b/db.go
@@ -244,7 +244,7 @@ func getYAML(prefix string, id string) ([]byte, error) {
 		content, err := ioutil.ReadFile(path)
 		return content, err
 	}
-	return nil, errors.New("Failed to find report with specified ID")
+	return nil, errors.New("failed to find report with specified ID")
 }
 
 //
@@ -459,8 +459,7 @@ func getStates() ([]PuppetState, error) {
 
 		// Percentage has to be capped :)
 		if total != 0 {
-			var c float64
-			c = float64(states[name])
+			c := float64(states[name])
 			tmp.Percentage = (c / float64(total)) * 100
 		}
 		data = append(data, tmp)
@@ -585,7 +584,7 @@ func getHistory() ([]PuppetHistory, error) {
 		var d string
 		err = rows.Scan(&d)
 		if err != nil {
-			return nil, errors.New("Failed to scan SQL")
+			return nil, errors.New("failed to scan SQL")
 		}
 
 		dates = append(dates, d)
@@ -630,7 +629,7 @@ func getHistory() ([]PuppetHistory, error) {
 
 			err = rows.Scan(&name, &count)
 			if err != nil {
-				return nil, errors.New("Failed to scan SQL")
+				return nil, errors.New("failed to scan SQL")
 			}
 			if name == "changed" {
 				x.Changed = count

--- a/db_test.go
+++ b/db_test.go
@@ -386,7 +386,7 @@ func TestMissiongReport(t *testing.T) {
 
 	_, err := getYAML("", "")
 
-	reg, _ := regexp.Compile("Failed to find report with specified ID")
+	reg, _ := regexp.Compile("failed to find report with specified ID")
 	if !reg.MatchString(err.Error()) {
 		t.Errorf("Got wrong error: %v", err)
 	}

--- a/yaml_parser.go
+++ b/yaml_parser.go
@@ -115,7 +115,7 @@ func parseHost(y *simpleyaml.Yaml, out *PuppetReport) error {
 	//
 	host, err := y.Get("host").String()
 	if err != nil {
-		return errors.New("Failed to get 'host' from YAML")
+		return errors.New("failed to get 'host' from YAML")
 	}
 
 	//
@@ -123,7 +123,7 @@ func parseHost(y *simpleyaml.Yaml, out *PuppetReport) error {
 	//
 	reg, _ := regexp.Compile("^([a-z0-9._-]+)$")
 	if !reg.MatchString(host) {
-		return errors.New("The submitted 'host' field failed our security check")
+		return errors.New("the submitted 'host' field failed our security check")
 	}
 
 	out.Fqdn = host
@@ -141,7 +141,7 @@ func parseTime(y *simpleyaml.Yaml, out *PuppetReport) error {
 	//
 	at, err := y.Get("time").String()
 	if err != nil {
-		return errors.New("Failed to get 'time' from YAML")
+		return errors.New("failed to get 'time' from YAML")
 	}
 
 	// Strip any quotes that might surround the time.
@@ -170,7 +170,7 @@ func parseStatus(y *simpleyaml.Yaml, out *PuppetReport) error {
 	//
 	state, err := y.Get("status").String()
 	if err != nil {
-		return errors.New("Failed to get 'status' from YAML")
+		return errors.New("failed to get 'status' from YAML")
 	}
 
 	switch state {
@@ -178,7 +178,7 @@ func parseStatus(y *simpleyaml.Yaml, out *PuppetReport) error {
 	case "unchanged":
 	case "failed":
 	default:
-		return errors.New("Unexpected 'status' - " + state)
+		return errors.New("unexpected 'status' - " + state)
 	}
 
 	out.State = state
@@ -277,7 +277,7 @@ func parseResources(y *simpleyaml.Yaml, out *PuppetReport) error {
 func parseLogs(y *simpleyaml.Yaml, out *PuppetReport) error {
 	logs, err := y.Get("logs").Array()
 	if err != nil {
-		return errors.New("Failed to get 'logs' from YAML")
+		return errors.New("failed to get 'logs' from YAML")
 	}
 
 	var logged []string
@@ -314,7 +314,7 @@ func parseLogs(y *simpleyaml.Yaml, out *PuppetReport) error {
 func parseResults(y *simpleyaml.Yaml, out *PuppetReport) error {
 	rs, err := y.Get("resource_statuses").Map()
 	if err != nil {
-		return errors.New("Failed to get 'resource_statuses' from YAML")
+		return errors.New("failed to get 'resource_statuses' from YAML")
 	}
 
 	var failed []Resource
@@ -405,7 +405,7 @@ func ParsePuppetReport(content []byte) (PuppetReport, error) {
 	//
 	yaml, err := simpleyaml.NewYaml(content)
 	if err != nil {
-		return x, errors.New("Failed to parse YAML")
+		return x, errors.New("failed to parse YAML")
 	}
 
 	//

--- a/yaml_parser_test.go
+++ b/yaml_parser_test.go
@@ -27,7 +27,7 @@ func TestBogusYaml(t *testing.T) {
 	//
 	// Ensure the error is what we expect.
 	//
-	reg, _ := regexp.Compile("Failed to parse YAML")
+	reg, _ := regexp.Compile("failed to parse YAML")
 	if !reg.MatchString(err.Error()) {
 		t.Errorf("Got wrong error: %v", err)
 	}
@@ -95,9 +95,7 @@ func TestHostName(t *testing.T) {
 		//
 		// Build up YAML
 		//
-		var tmp string
-
-		tmp = "---\n" +
+		tmp := "---\n" +
 			"host: " + input.hostname
 
 		//
@@ -162,9 +160,7 @@ func TestNodeStatus(t *testing.T) {
 		//
 		// Build up YAML
 		//
-		var tmp string
-
-		tmp = "---\n" +
+		tmp := "---\n" +
 			"host: foo.example.com\n" +
 			"time: '2017-08-07T16:37:42.659245699+00:00'\n" +
 			"status: " + input.state

